addons/trenches/functions/fnc_initTrench.sqf

```diff
@@ -0,0 +1,35 @@
+#include "script_component.hpp"
+/*
+ * Author: chris579, Salbei
+ * Inits a trench
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [TrenchObj] call ace_trenches_fnc_initTrench
+ *
+ * Public: No
+ */
+
+params [
+    ["_object", objNull, [objNull]]
+];
+if (isServer) then {
+   _object setVariable [QGVAR(progress), 1, true];
+};
+
+if (is3DEN) exitWith {
+    [_object] call FUNC(initTrench3DEN);
+};
+
+if (local _object) then {
+    // Has to be spawned to ensure MP compatibility
+    _object spawn {
+        private _texture = [_this] call FUNC(getSurfaceTexturePath);
```

Comment:
_this shouldn't need to be packed into an array

Reply:
I just find it easier to read, this is just my subjectiv opinion. If this couses any performance issues, it can be changed.